### PR TITLE
Add ingressmonitorcontroller

### PR DESCRIPTION
**Pull request by faizanahmad055**
Signed-off-by: Faizan Ahmad
username: faizanahmad055
email: faizan.ahmad55@outlook.com

#### What this PR does / why we need it:
- This PR adds the official chart for [IngressMonitorController](https://github.com/stakater/IngressMonitorController)
- IngressMonitorController continuously watches ingresses/routes in specific or all namespaces, and automatically add/remove monitors in any of the uptime checkers.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/ingressmonitorcontroller/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+kind: Deployment
+metadata:
+  annotations:
+    configmap.reloader.stakater.com/reload: {{ template "name" . }}
+  labels:
+    app: {{ template "name" . }}
+    version: {{ .Chart.Version }}
+    group: {{ .Values.ingressMonitorController.labels.group }}
+    provider: {{ .Values.ingressMonitorController.labels.provider }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
Comment (davidkarlsen) [inline stable/ingressmonitorcontroller/templates/deployment.yaml:10]: See best practice label usage at: https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

File: stable/ingressmonitorcontroller/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
+apiVersion: v1
+kind: ServiceAccount
```
Comment (davidkarlsen) [inline stable/ingressmonitorcontroller/templates/rbac.yaml]: See RBAC best practices at: https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md

File: stable/ingressmonitorcontroller/templates/rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    version: {{ .Chart.Version }}
+    group: {{ .Values.ingressMonitorController.labels.group }}
+    provider: {{ .Values.ingressMonitorController.labels.provider }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+  name: {{ template "name" . }}
+---
+{{- if .Values.ingressMonitorController.watchNamespace }}
+apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (davidkarlsen) [inline stable/ingressmonitorcontroller/templates/rbac.yaml]: can use api version 1

File: stable/ingressmonitorcontroller/templates/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  - apiGroups:
+      - ""
+    resources:
+      - services
+      - pods
+    verbs:
+      - list
+      - get
+      - watch
+---
```
Comment (davidkarlsen) [inline stable/ingressmonitorcontroller/templates/rbac.yaml]: the role, rolebinding and clusterrolebinding should be split into separate template-files.

Comment (davidkarlsen) [reply stable/ingressmonitorcontroller/templates/rbac.yaml]: @faizanahmad055 ping

File: stable/ingressmonitorcontroller/templates/deployment.yaml (hunk 1 above)
Comment (faizanahmad055) [reply stable/ingressmonitorcontroller/templates/deployment.yaml:10]: Updated.

File: stable/ingressmonitorcontroller/templates/rbac.yaml (hunk 2 above)
Comment (faizanahmad055) [reply stable/ingressmonitorcontroller/templates/rbac.yaml]: Updated

File: stable/ingressmonitorcontroller/templates/rbac.yaml (hunk 3 above)
Comment (faizanahmad055) [reply stable/ingressmonitorcontroller/templates/rbac.yaml]: Updated.

File: stable/ingressmonitorcontroller/templates/rbac.yaml (hunk 4 above)
Comment (faizanahmad055) [reply stable/ingressmonitorcontroller/templates/rbac.yaml]: Hi sorry for the delay, I've updated these. Can you please take a look.

**Comment by faizanahmad055** [comment]
@davidkarlsen 
Ping

File: stable/ingressmonitorcontroller/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
+apiVersion: extensions/v1beta1
```
Comment (davidkarlsen) [inline stable/ingressmonitorcontroller/templates/deployment.yaml]: can be v1

File: stable/ingressmonitorcontroller/templates/role.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+{{- if and ( .Values.ingressMonitorController.rbac.create ) (.Values.ingressMonitorController.watchNamespace) }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
```
Comment (davidkarlsen) [inline stable/ingressmonitorcontroller/templates/role.yaml:4]: add labels

File: stable/ingressmonitorcontroller/templates/clusterrolebinding.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+{{- if and ( .Values.ingressMonitorController.rbac.create ) (eq .Values.ingressMonitorController.watchNamespace "") }}
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
```
Comment (davidkarlsen) [inline stable/ingressmonitorcontroller/templates/clusterrolebinding.yaml:4]: add labels

File: stable/ingressmonitorcontroller/templates/clusterrole.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
+{{- if and ( .Values.ingressMonitorController.rbac.create ) (eq .Values.ingressMonitorController.watchNamespace "") }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+{{- if .Values.ingressMonitorController.useFullName }}
```
Comment (davidkarlsen) [inline stable/ingressmonitorcontroller/templates/clusterrole.yaml:5]: add labels

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: stable/ingressmonitorcontroller/templates/clusterrole.yaml (hunk 8 above)
Comment (faizanahmad055) [reply stable/ingressmonitorcontroller/templates/clusterrole.yaml:5]: Updated

File: stable/ingressmonitorcontroller/templates/deployment.yaml (hunk 5 above)
Comment (faizanahmad055) [reply stable/ingressmonitorcontroller/templates/deployment.yaml]: Updated

File: stable/ingressmonitorcontroller/templates/role.yaml (hunk 6 above)
Comment (faizanahmad055) [reply stable/ingressmonitorcontroller/templates/role.yaml:4]: Updated

File: stable/ingressmonitorcontroller/templates/clusterrolebinding.yaml (hunk 7 above)
Comment (faizanahmad055) [reply stable/ingressmonitorcontroller/templates/clusterrolebinding.yaml:4]: Updated

**Comment by faizanahmad055** [comment]
Hi @davidkarlsen,
Ping

**Comment by paulczar** [comment]
@davidkarlsen looks like this has implemented your requested changes ... I think it looks mergable now, but I'll give you a few more days to LGTM it before I step in.

**Review by davidkarlsen — APPROVED** [review]

**Pull request merged**